documentation/performance/sizing-feeding.html

```diff
@@ -0,0 +1,249 @@
+</ol>
+Then see next section
+</p>
+
+
+
+<h2 id="feed-testing">Feed testing</h2>
+<p>
+When testing for feeding capacity:
+<ol>
+  <li>Use the <a href="../vespa-http-client.html">Java HTTP client</a> in asynchronous mode.</li>
+  <li>Test feeding performance by adding feeder instances.
+    Make sure network and CPU (content and container node) usage increases, until saturation.</li>
+  <li>See troubleshooting at end to make sure there are no errors</li>
+</ol>
+Scenarios: Feed testing for capacity for sustained load in a system in steady state,
+during state changes, during query load.
+</p>
+
+
+<h2 id="troubleshooting">Troubleshooting</h2>
+<p>
+Find a non-exhaustive list of things to check below:
+</p>
+
+<h3 id="metrics">Metrics</h3>
+<p>
+Look at queues on content nodes - queue wait time and queue size:
+<pre>
+vds.filestor.alldisks.averagequeuewait.sum average=217.635 last=94.464 min=6.6731 max=598.772 count=80554 total=1.75313e+07
+vds.filestor.alldisks.queuesize average=11.95 last=3 min=0 max=119 count=60 total=717
+</pre>
+Check content node metrics across all nodes to see if there are any outliers. Also check
+<pre>
+vds.filestor.alldisks.allthreads.update.sum.latency
+</pre>
+</p>
+
+
+<h3 id="failure-rates">Failure rates</h3>
+<p>
+<pre>
+vds.distributor.updates.sum.latency average=6324.93 last=4480.5 min=2.06559 max=35757.4 count=1596 total=1.00946e+07
+vds.distributor.updates.sum.ok count=1101
+vds.distributor.updates.sum.failures.total count=1085
+vds.distributor.updates.sum.failures.storagefailure count=495
+</pre>
+</p>
+
+
+<h3 id="blocked-feeding">Blocked feeding</h3>
+<p>
+Should be 0 everywhere -
+refer to <a href="../writing-to-vespa.html#feed-block">feed block</a>:
+<pre>
+content.proton.resource_usage.feeding_blocked
+</pre>
+
+</p>
+
+
+<h3 id="concurrent-mutations">Concurrent mutations</h3>
+<p>
+<pre>
+vds.distributor.updates.sum.failures.concurrent_mutations count=1484
+</pre>
+Mutating client operations towards a given document ID are sequenced on the distributors.
+If an operation is already active towards a document,
+a subsequently arriving one will be bounced back to the client with a transient failure code.
+Usually this happens when users send feed from multiple clients concurrently without synchronisation.
+Note that feed operations sent by a single client are sequenced client-side,
+so this should not be observed with a single client only.
+Bounced operations are never sent on to the backends and should not cause elevated latencies there,
+although the client will observe higher latencies due to automatic retries with back-off.
+</p>
+
+
+<h3 id="wrong-distribution">Wrong distribution</h3>
+<p>
+<pre>
+vds.distributor.updates.sum.failures.wrongdistributor count=590
+</pre>
+Indicates that clients keep sending to the wrong distributor.
+Normally this shouldn’t happen,
+as clients update and cache all state required to route directly to the correct distributor
```

Comment:
Consider rephrasing as "Normally this should happen infrequently" instead of "Normally this shouldn't happen", as it _does_ happen on client startup or distributor state transitions.